constraints.txt

```diff
@@ -6,3 +6,7 @@ jsonschema==3.2.0
 # a seaborn import. This pin can be removed when compatibility with those
 # packages is fixed
 jupyter-core==4.11.2
+
+# ipywidgets 8.0.3 started emitting deprecation warnings via a seaborn import.
+# This pin can be removed when compatibility with those packages is fixed.
+ipywidgets<8.0.3
```

Comment:
Was there a case just pinning in the requirements list wasn't enough and we needed to pin it in the constraints file too?

Reply:
This line upgrades ipywidgets to 8.0.3 without this change of constraints.txt.
https://github.com/Qiskit/qiskit-terra/blob/ee4a5712bfac6104e3e93fc8a3ff845a3a0ddd77/.azure/test-linux.yml#L156

Reply:
We probably don't need the requirements-dev.txt change in parallel then. But it doesn't hurt anything either.

Reply:
Ah, you are right... I didn't have to change in parallel.